### DISCUSSION TITLE
Im getting error when trying to import a reel

### DISCUSSION THREAD

**Discussion by GerardPolloRebozado** [JoTec2002/InstagramToMealie#5] [Q&A]
`xception('Error while getting Recipe from API! - Status Code: 400 - Response: {"detail":{"details":"BAD_RECIPE_DATA"}}')`
Do I have anything wrong or reels are not supported?

**Comment by JoTec2002** [comment]
Reals in generell work.

How do you have the OpenAI konfigured in Mealie?
Which model are you using?
Have you tried it maybe with an different real?

**Reply by GerardPolloRebozado** [reply]
I have configured groq with llama3

OPENAI_BASE_URL: https://api.groq.com/openai/v1
 OPENAI_API_KEY:  xx
 OPENAI_ENABLE_IMAGE_SERVICES: false
   OPENAI_MODEL: llama3-8b-8192

I think it's an ai problem tried with 5 more reels 1 of them worked so probably is the ai that is not very good.
Are there any guidelines on how should the reel be to get better results? Does language matter? Is the description of the video used?

**Reply by JoTec2002** [reply]
The complete description of the instagram reel is feed to the LLM in addition to the prompt from Mealie. 
Personally I found that some LLM Models don't return a correctly formatted json. (That's where your error comes from) I got some great results with qwen2.5:7b, their in 90% of the time correct.

Since I'm using the Mealie API import for HTML/ whole website the system prompt given to the LLM might not be optimal. There is already an Pull Request in the Mealie Repo addressing this by providing a new endpoint for just text input with improved System Prompt: https://github.com/mealie-recipes/mealie/pull/4561 As soon as this is merged I will switch the instagram importer to the new endpoint.

**Reply by GerardPolloRebozado** [reply]
thanks for the reply and this awesome project. as a suggestion I think that after importing the meal it would be nice to be redirected to the added receipe. This could be implemented like in mealie that you have a check to say if you want to edit after importing the receipe

**Reply by GerardPolloRebozado** [reply]
> The complete description of the instagram reel is feed to the LLM in addition to the prompt from Mealie. Personally I found that some LLM Models don't return a correctly formatted json. (That's where your error comes from) I got some great results with qwen2.5:7b, their in 90% of the time correct.
> 
> Since I'm using the Mealie API import for HTML/ whole website the system prompt given to the LLM might not be optimal. There is already an Pull Request in the Mealie Repo addressing this by providing a new endpoint for just text input with improved System Prompt: [mealie-recipes/mealie#4561](https://github.com/mealie-recipes/mealie/pull/4561) As soon as this is merged I will switch the instagram importer to the new endpoint.

I have been trying with the qwen2.5:7b and also qwen2.5:32b:instruct on hugging face and most of the time im getting bad receipe data

**Reply by JoTec2002** [reply]
Can you maybe provide some examples on which the import fails

**Reply by GerardPolloRebozado** [reply]
Sure here are some:
https://www.instagram.com/p/DCx_ohkthZO/
https://www.instagram.com/p/DDVzUEXvstJ/
https://www.instagram.com/p/DBbCXYfIKr7/

**Reply by JoTec2002** [reply]
Tried the first one and worked without any Problems.
I'm running qwen2.5:7b via ollama, I don't know if this makes a difference. 

Maybe Try importing the instagram discription text via the Mealie Web UI as HTML and see if this works as a Troubleshooting step.

**Reply by GerardPolloRebozado** [reply]
Do you mean this? It doesnt do anything when making click in create
![image](https://github.com/user-attachments/assets/14fedd0e-e265-4c0a-b788-70969868773f)
I also tried using together ai for the model of Qwen2.5 and same issue

**Reply by JoTec2002** [reply]
Yeah that's the import I meant.

Maybe then try to check in Mealie, if your OpenAI configuration is working `/admin/debug/openai`.

**Reply by GerardPolloRebozado** [reply]
This is the reply I get:
OpenAI is working. Response: "It seems like you've sent an object, but no specific details were provided. Could you please provide more context or details about what you're trying to debug or understand?"

**Reply by JoTec2002** [reply]
> Do you mean this? It doesnt do anything when making click in create ![image](https://private-user-images.githubusercontent.com/37554513/394914106-14fedd0e-e265-4c0a-b788-70969868773f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5OTA0MTIsIm5iZiI6MTczMzk5MDExMiwicGF0aCI6Ii8zNzU1NDUxMy8zOTQ5MTQxMDYtMTRmZWRkMGUtZTI2NS00YzBhLWI3ODgtNzA5Njk4Njg3NzNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDA3NTUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2OTllYmM3YzFjZWY0ZGEzMjQxN2M2OWQzZDMwMjRhNDk0MzVlYmJmN2Y1MjNmNmExNzlmMjc2YjdhNjkwNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1pKO8gDVNgzcEiM4CAAbH6qqJ7ALEja6jtIQfvTKteM) I also tried using together ai for the model of Qwen2.5 and same issue

If this doesn't work in Mealie maybe thats the root cause you having issues. Which version of Mealie are you using?

**Reply by GerardPolloRebozado** [reply]
I have mealie 2.3.0 setting says it's up-to-date ![Screenshot_20241212-101323.png](https://github.com/user-attachments/assets/7f8f7b49-6868-4e7b-bda9-042338f6d23c)

While checking settings also saw an option to check the ai parser works
![Screenshot_20241212-102038.png](https://github.com/user-attachments/assets/ff8152dd-3c8e-4b7c-991d-cd58f8b953b1)

**Reply by JoTec2002** [reply]
> While checking settings also saw an option to check the ai parser works

This parses the ingredients of an already existing recipie from my understanding.


I'm really sorry but I'm out of ideas. The only thing you could try would be trying different models until you find one that functions for you consistently.
Mealie is adding the following infornt of your input/ the instagram discription https://github.com/mealie-recipes/mealie/blob/mealie-next/mealie/services/openai/prompts/recipes/scrape-recipe.txt maybe you could do some example prompts with that to simplyfy model testing.
Still waiting on upstream pull request https://github.com/mealie-recipes/mealie/pull/4561 to be finished and merged, that could help you also.

**Reply by GerardPolloRebozado** [reply]
Ok thanks for your time I will test done things and post here any solution I find. When receiving the prompt what does the ai receive apart from the description? Is the audio from the reel converted to text with whisper?

**Reply by JoTec2002** [reply]
Sorry for the late response. The AI only gets the description from the Instagram post/reel.

**Comment by applesoff** [comment]
i am getting an error when trying to import an IG reel too. i have tried many. used a few ollama models too.
logs just show below: 
mealie-instagramtomealie-1  | downloads/DEjAqn7Mihg/2025-01-08_02-00-18_UTC.jpg [Marinated Sun-Dried Tomato Be…] downloads/DEjAqn7Mihg/2025-01-08_02-00-18_UTC.mp4 
mealie-instagramtomealie-1  | Downloaded post: https://www.instagram.com/p/DEjAqn7Mihg/
mealie                      | ERROR    2025-01-19T01:31:08 - Recipe Scraper was unable to extract a recipe from 
mealie                      | ERROR    2025-01-19T01:31:08 - Recipe Scraper was unable to extract a recipe from 
mealie                      | INFO     2025-01-19T01:31:39 - [127.0.0.1:45490] 200 OK "GET /api/app/about HTTP/1.1"

and the error i get on the page is: ReadTimeout(ReadTimeoutError("HTTPSConnectionPool(host='mealie.pendra.dev', port=443): Read timed out. (read timeout=60)"))

or

Exception('Error while getting Recipe from API! - Status Code: 400 - Response: {"detail":{"details":"BAD_RECIPE_DATA"}}')

**Reply by JoTec2002** [reply]
@applesoff i got a notification, that you also posted something in Discussion #1. 

There at least you posted you're docker-compose config and some more extend logs.

It seams going from that, that this converter can't connect to mealie. Please make sure your MEALIE_URL environment variable starts with http:// or https://.

**Reply by applesoff** [reply]
Yeah I figured it out. The issue above is what I am currently stuck on. It's fails every time with either of these two issues

**Reply by JoTec2002** [reply]
Since I don't know on which hardware you're running the model and there are timeout issues have you tried increasing the timeout (in mealie itself and in this importer)?
Have you verified that the OpenAI integration in Mealie  works?
